src/docs/patterns/tables.mdx

```diff
@@ -0,0 +1,463 @@
+        `When a row is selected, the application enters this "batch action mode", and users can select other rows by clicking anywhere on those rows.`,
+    ]}
+/>
+-->
+
+## Sticky Header Row
+
+When a table gets long, your users should not need to scroll back to the top to view the table header.
+
+<ImageGrid
+    images={[StickyHeader]}
+    caption={[`Use sticky headers to help your users navigate through the table.`]}
+    regularWidth
+/>
+
+## Expanded Row
+
+Occasionally, table rows may expand to allow users _take a quick peek_ into a small amount of row details.
+
+<ImageGrid
+    images={[ExpandedRow]}
+    caption={[`Users can quickly glance into a small amount of row details.`]}
+    regularWidth
+/>
+
+<ImageGrid
+    images={[ExpandedRowTooComplicated, ExpandedRowActions]}
+    captionsUnderImages={[
+        `DONT: Do not include complicated contents in an expanded row, as this will break the efficiency of a table. In this case, you should design a dedicated page for the content.`,
+        `DONT: Do not surprise users by hiding high priority actions in the expanded row space.`,
+    ]}
+/>
+
+## Horizontal Scrolling
+
+When your table overflows horizontally, you should allow your users to scroll horizontally to view more about the table. However, this interaction is less ergonomically convenient when users are not using a touch screen or a trackpad. See the [Responsive](#responsive) section below.
+
+The application may display a scroll bar to suggest that there are more contents to the right.
+
+<ImageGrid
+    images={[HorizontalScrollBar]}
+    caption={[`Place horizontal scroll bar between rows and the footer / pagination area.`]}
+    regularWidth
+/>
+
+<ImageGrid
+    images={[StickyCheckbox, StickyRowAction]}
+    captionsUnderImages={[
+        `The checkbox column should be sticky to the left as the user scrolls to the right.`,
+        `The action column, if designed to be always visible, should be sticky to the right. Do not include too many row actions that cover excessive table body area.`,
+    ]}
+/>
+
+## Searching and Filtering
+
+There are many different ways to perform searching and filtering with table data. Below are a few examples:
+
+<ImageGrid
+    images={[Filter1, Filter2, Filter3, Filter4]}
+    captionsUnderImages={[
+        `A simple search field embedded in a table.`,
+        `A search field outside a table.`,
+        `Filter criteria listed as chips.`,
+        `A "saved filter" dropdown.`,
+    ]}
+/>
+
+<ImageGrid
+    images={[AdvancedTableFilter]}
+    caption={[
+        `You may combine multiple interactions (searching and filtering) and input methods (keyboard navigations, keyboard typing, mouse clicking) together for an "advanced search".`,
+    ]}
+    regularWidth
+/>
+
+## Transition Animation
+
+Often, table data look highly similar between rows and can disorient your users. You are encouraged to use animations in your applications to help users understand certain transitions.
```

Comment:
```suggestion
Table rows often look very similar and can be difficult for users to differentiate. You should make use of animations and transitions to help users understand where certain actions are taking place.
```

Reply:
Since you changed it from "we encourage" to "you should", I am going to add `DO:` to the image captions to emphasize that this is a must.